SYCL/Reduction/reduction_aux_resources.cpp

```diff
@@ -0,0 +1,35 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_60
```

Comment:
the test is built for CUDA BE
Is it expected? 
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```

Reply:
These options only have an effect if the sycl target is the CUDA backend and are ignored otherwise. All it does is instruct the generation of NVPTX to use SM60, which I believe is required here to enable an extended set of atomic operations used by reductions.